server/src/test/java/org/elasticsearch/reservedstate/service/FileSettingsServiceTests.java

```diff
@@ -223,40 +227,40 @@ public void testProcessFileChanges() throws Exception {
             return null;
         }).when(controller).process(any(), any(XContentParser.class), any(), any());
 
-        // we get three events: initial clusterChanged event, first write, second write
```

Comment:
This was not a very robust setup... The two things we really care about is that `processFileOnServiceStart` gets called, and that `processFileChanges` and that we wait for these calls. I've restructured this test to use two separate latches, and do the file writes one at a time.